### PR TITLE
Topology, fetch build config data when necessary

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5114

**Solution Description**: 
Remove build configs from OverviewItem type. Add hook to retrieve build configs for a given resource. Update instances to use hooks rather than fields in the OverviewItem type.

The alerts field for OverviewItem type is also removed as it is not being used anywhere (but required loading buildConfig resources).

Non-Functional Change

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

/kind cleanup

### CODE REVIEW

File: frontend/packages/console-shared/src/hooks/useBuildConfigsWatcher.ts
Comment (jerolimov) [inline frontend/packages/console-shared/src/hooks/useBuildConfigsWatcher.ts:37]: Small detail, but can simplified with Object.values:

```suggestion
    const resourceWithLoadError = Object.values(resources).find((r) => r.loadError);
    if (resourceWithLoadError) {
      setLoadError(resourceWithLoadError.loadError);
      return;
    }
```

For me both is fine, doesn't block this PR for me.

File: frontend/packages/console-shared/src/hooks/useBuildConfigsWatcher.ts
Comment (jerolimov) [inline frontend/packages/console-shared/src/hooks/useBuildConfigsWatcher.ts:11]: Please default with null. The useEffect also sets setLoadError to null if there is new resources available without any error. In this case we can save us from one re-rendering

File: frontend/packages/dev-console/src/utils/application-utils.ts
Comment (jerolimov) [inline frontend/packages/dev-console/src/utils/application-utils.ts:271]: Just an idea: We can move this check also into `getBuildConfigsForResource` ? So that it starts with

```tsx
export const getBuildConfigsForResource = (
  resource: K8sResourceKind,
  resources: any,
): BuildConfigOverviewItem[] => {
  if (resource.kind === 'CronJob') {
    return getBuildConfigsForCronJob(resource as CronJobKind, resources);
  }
  // ....
```

Makes this sense? Not required to get a lgtm :)

**Review by jerolimov — CHANGES_REQUESTED** [review]
Finished review, small change requested and added two ideas.

Tested the topology graph and sidebar locally with some git deployment which works fine.

**Review by jerolimov — APPROVED** [review]
Tested locally, Topology shows BuildConfigs

/lgtm

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Pull request merged**
